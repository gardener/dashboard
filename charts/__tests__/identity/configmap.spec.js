//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const yaml = require('js-yaml')
const { omit } = require('lodash')
const { helm } = fixtures

const renderTemplates = helm.renderIdentityTemplates

describe('identity', function () {
  describe('configmap', function () {
    let templates

    beforeEach(() => {
      templates = [
        'configmap'
      ]
    })

    it('should render the template w/ defaults values', async function () {
      const documents = await renderTemplates(templates, {})
      expect(documents).toHaveLength(1)
      const [configMap] = documents
      expect(omit(configMap, ['data'])).toMatchSnapshot()
      expect(Object.keys(configMap.data)).toEqual(['config.yaml'])
      const config = yaml.load(configMap.data['config.yaml'])
      expect(config).toMatchSnapshot()
    })
  })
})
