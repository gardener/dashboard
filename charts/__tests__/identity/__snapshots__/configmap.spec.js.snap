// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identity configmap should render the template w/ defaults values 1`] = `
{
  "apiVersion": "v1",
  "kind": "ConfigMap",
  "metadata": {
    "name": "identity-configmap",
    "namespace": "garden",
  },
}
`;

exports[`identity configmap should render the template w/ defaults values 2`] = `
{
  "connectors": [
    {
      "config": {
        "caData": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCkxpNHUKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQ==",
        "emailAttr": "email",
        "entityIssuer": "identity.ingress.example.org",
        "groupsAttr": "groups",
        "nameIDPolicyFormat": "unspecified",
        "redirectURI": "https://identity.ingress.example.org/callback",
        "ssoIssuer": "saml.example.org",
        "ssoURL": "https://saml.example.org/sso",
        "usernameAttr": "name",
      },
      "id": "saml",
      "name": "SAML",
      "type": "saml",
    },
  ],
  "enablePasswordDB": false,
  "issuer": "https://identity.ingress.example.org",
  "oauth2": {
    "responseTypes": [
      "token",
      "code",
      "id_token",
    ],
    "skipApprovalScreen": true,
  },
  "staticClients": [
    {
      "id": "dashboard",
      "name": "Gardener Dashboard",
      "redirectURIs": [
        "https://dashboard.garden.example.org/auth/callback",
        "https://dashboard.ingress.garden.example.org/auth/callback",
        "http://localhost:8080/auth/callback",
      ],
      "secret": "sHq4vLoiQcIWbO3h",
    },
    {
      "id": "kube-kubectl",
      "name": "Kubectl",
      "public": true,
      "secret": "if6ji0dTFE4rQfj8",
      "trustedPeers": [
        "dashboard",
      ],
    },
  ],
  "staticPasswords": [],
  "storage": {
    "config": {
      "inCluster": true,
    },
    "type": "kubernetes",
  },
  "web": {
    "allowedOrigins": [
      "https://dashboard.garden.example.org",
      "https://dashboard.ingress.garden.example.org",
      "http://localhost:8080",
    ],
    "http": "0.0.0.0:5556",
  },
}
`;
